### ISSUE TITLE
[YSQL] tserver crashes during insert if indexed column is null

### ISSUE THREAD

**Issue by ndeodhar** [YugaByte/yugabyte-db#945]
To reproduce:
```
postgres=# create table foo(a int, b int, c int, d int);
CREATE TABLE
postgres=# create index foo_a on foo(a);
CREATE INDEX
postgres=# create index foo_bc on foo(b, c);
CREATE INDEX
postgres=# insert into foo values(1,1,1,1);
INSERT 0 1
postgres=# insert into foo(b,c,d) values (2,2,2);
ERROR:  Timed out: Write(tablet: bb0786ab59e84c768863dfe68bae550e, num_ops: 1, num_attempts: 342, txn: c4c6ad1c-6bea-4c8c-a756-b4d8aa59601b) passed its deadline 294786.908s (now: 294786.922s): Network error (yb/util/net/socket.cc:572): Recv() got EOF from remote (error 58)
```

**Comment by robertpang** [comment]
tserver encountered an unexpected value type 88:

```
F0314 21:46:09.257661 116940800 enums.h:176] Invalid value of enum ValueType (full enum type: yb::docdb::ValueType, expression: type_): 88.
```

**Comment by robertpang** [comment]
The issue is fixed in commit [1dedcd1](https://github.com/YugaByte/yugabyte-db/commit/1dedcd1f4481d50a32517f3af2b4d5e188abf17d).

**Comment by frozenspider** [comment]
Related issue: #1058
